### PR TITLE
Rollup up 3 pull requests

### PR DESCRIPTION
**Pull request by matthiaskrgr**
changelog: none

We need all of these merging at the same time.

### CODE REVIEW

**Comment by matthiaskrgr** [comment]
```
find: ‘/home/runner/work/rust-clippy/rust-clippy/target/debug/incremental/cargo_clippy-hfbbgcctcpct’: No such file or directory
mv: cannot stat '/home/runner/work/rust-clippy/rust-clippy/target/debug/integration-*': No such file or directory
##[error]Process completed with exit code 1.

```
looks spurious :thinking: 

@bors retry

**Comment by matthiaskrgr** [comment]
````
    # Upload
    - name: Extract Binaries
      run: |
        DIR=$CARGO_TARGET_DIR/debug
        find $DIR ! -executable -o -type d ! -path $DIR | xargs rm -rf
        mv $DIR/integration-* $DIR/integration
        rm -rf $CARGO_TARGET_DIR/release

    - name: Upload Binaries
      uses: actions/upload-artifact@v1
      with:
        name: target
        path: target

````
`$DIR` seems to be `target/debug` so we are looking for `target/debug/integration-*` which does not seem to exist?

**Comment by flip1995** [comment]
`integration-*` is an executable file, where `*` is a hash code. Maybe if tests get compiled, without executing, the suppress the hash now?

For debugging you could add 

```
- name: Debug
  run: ls target/debug -l
```

before the extract binary step. This will show which files get produced by compiling the integration test.

**Comment by flip1995** [comment]
```
drwxr-xr-x 30 runner docker      4096 Mar 19 11:52 build
-rwxr-xr-x  2 runner docker   3591384 Mar 19 11:55 cargo-clippy
-rwxr-xr-x  2 runner docker 376728760 Mar 19 11:55 clippy-driver
drwxr-xr-x  2 runner docker    139264 Mar 19 11:55 deps
drwxr-xr-x  2 runner docker      4096 Mar 19 11:52 examples
drwxr-xr-x  9 runner docker      4096 Mar 19 11:55 incremental
```
yeah, somehow the test binary is not produced.

File: tests/ui/transmute_ptr_to_ptr.stderr [hunk 1]
```diff
@@ -1,3 +1,17 @@
+error: transmute from a type (`&T`) to itself
+  --> $DIR/transmute_ptr_to_ptr.rs:8:5
+   |
+LL |     std::mem::transmute::<&'a T, &'static T>(t)
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::useless-transmute` implied by `-D warnings`
```
Comment (flip1995) [inline tests/ui/transmute_ptr_to_ptr.stderr]: We explicitly **don't** want to lint this. See my comment in https://github.com/rust-lang/rust-clippy/pull/5333#discussion_r394277171

**Comment by flip1995** [comment]
cc rust-lang/cargo#7965

**Comment by flip1995** [comment]
This should fix it:

```diff
diff --git a/.github/workflows/clippy_bors.yml b/.github/workflows/clippy_bors.yml
index a4f7b64c..8393b16d 100644
--- a/.github/workflows/clippy_bors.yml
+++ b/.github/workflows/clippy_bors.yml
@@ -209,7 +209,7 @@ jobs:
       run: |
         DIR=$CARGO_TARGET_DIR/debug
         find $DIR ! -executable -o -type d ! -path $DIR | xargs rm -rf
-        mv $DIR/integration-* $DIR/integration
+        mv $DIR/deps/integration-* $DIR/integration
         rm -rf $CARGO_TARGET_DIR/release

     - name: Upload Binaries
```

**Comment by flip1995** [comment]
We'll need another rustup: #66131 

I'm working on it. I'll open a new rollup PR, once it's ready.

@bors treeclosed=1000

**Pull request closed without merging**